src/EditorFeatures/Core/RenameTracking/RenameTrackingTaggerProvider.cs

```diff
@@ -103,63 +104,44 @@ internal static bool ResetRenameTrackingStateWorker(Workspace workspace, Documen
 
     public static (CodeAction action, TextSpan renameSpan) TryGetCodeAction(
         Document document, TextSpan textSpan,
-            IEnumerable<IRefactorNotifyService> refactorNotifyServices,
-            ITextUndoHistoryRegistry undoHistoryRegistry,
-            CancellationToken cancellationToken)
+        IEnumerable<IRefactorNotifyService> refactorNotifyServices,
+        ITextUndoHistoryRegistry undoHistoryRegistry)
     {
         try
         {
             if (document != null && document.TryGetText(out var text))
             {
                 var textBuffer = text.Container.TryGetTextBuffer();
                 if (textBuffer != null &&
-                    textBuffer.Properties.TryGetProperty(typeof(StateMachine), out StateMachine stateMachine) &&
-                    stateMachine.CanInvokeRename(out _, cancellationToken: cancellationToken))
+                    textBuffer.Properties.TryGetProperty(typeof(StateMachine), out StateMachine stateMachine))
                 {
                     return stateMachine.TryGetCodeAction(
-                        document, text, textSpan, refactorNotifyServices, undoHistoryRegistry, cancellationToken);
+                        document, text, textSpan, refactorNotifyServices, undoHistoryRegistry);
                 }
             }
 
             return default;
         }
-        catch (Exception e) when (FatalError.ReportAndPropagateUnlessCanceled(e, cancellationToken, ErrorSeverity.General))
+        catch (Exception e) when (FatalError.ReportAndPropagateUnlessCanceled(e, ErrorSeverity.General))
         {
             throw ExceptionUtilities.Unreachable();
         }
     }
 
-    internal static bool IsRenamableIdentifier(Task<TriggerIdentifierKind> isRenamableIdentifierTask, bool waitForResult, CancellationToken cancellationToken)
+    internal static bool IsRenamableIdentifierFastCheck(
```

Comment:
💡 Consider renaming these back (removing the `FastCheck` suffix) since there is no corresponding "slow check"

Reply:
i liked the FastCheck as it indicated to the caller that this wasn't a guaranteed check, but just a quick one that could be innacurate.